inspirehep/modules/records/api.py

```diff
@@ -77,24 +80,92 @@ def create(cls, *args, **kwargs):
                 back to the value of the ``RECORDS_SKIP_FILES`` configuration
                 variable.
 
+        Examples:
+            >>> record = {
+            ...     '$schema': 'hep.json',
+            ... }
+            >>> record = InspireRecord.create(record)
+            >>> record.commit()
         """
-        config_skip_files = current_app.config.get('RECORDS_SKIP_FILES')
+        files_src_records = kwargs.pop('files_src_records', [])
+        skip_files = kwargs.pop(
+            'skip_files', current_app.config.get('RECORDS_SKIP_FILES'))
 
-        files_src_records = kwargs.pop('files_src_records', ())
-        skip_files = kwargs.pop('skip_files', config_skip_files)
+        id_ = id_ or uuid.uuid4()
+        data = strip_empty_values(data)
 
-        data = strip_empty_values(*args)
-        new_record = super(InspireRecord, cls).create(data=data, **kwargs)
+        cls.mint(id_, data)
+        record = super(InspireRecord, cls).create(data, id_=id_, **kwargs)
 
         if not skip_files:
-            new_record.download_documents_and_figures(
+            record.download_documents_and_figures(
                 src_records=files_src_records,
             )
-            new_record.commit()
 
-        return new_record
+        return record
+
+    @classmethod
+    def create_or_update(cls, data, **kwargs):
```

Comment:
conceptually, I would expect `create_or_update` to be something like:
```
try:
    update(...)
except PIDDoesNotExistError:
    create(...)
```
what is the reason it has to be much more complex than that? (I know you are not refactoring anything yet, those are hints for later).